highway.tex

```diff
@@ -615,9 +630,207 @@ \subsection{Checkpoints}
+
+
+\subsection{Spam Prevention Techniques}
+
+We begin by introducing some new definitions.
+
+%\subsubsection{Chains and Endorsements}
+
+\begin{definition}
+A \emph{chain} is a totally ordered sequence of messages by the same validator. Equivalently, it is a set of messages by a single validator that contains no pair of equivocating messages.
+\end{definition}
+
+Notice that if we limit the round exponents that can be used by any validators and require that they send no more than three sequential votes per round, this guarantees that no chain will contain too many messages. Our goal will thus be to ensure that we never need to include too many chains of messages in our justifications. We note that one equivocating validator might send equivocating votes to each of the other $n$ validators and if they do not coordinate about which votes to include in their next justification, they may be forced to include $n$ chains from this validator in order to incorporate each others' votes. However, we will want to ensure that the situation can not be made to be much worse than that by some kind of equivocation bomb, and this will require that the honest validators coordinate on which messages to include. We do this using a system of endorsements described below.
+
+\begin{definition}
+An \emph{endorsement} is a non-vote message between validators that consists of a validator $V$ and the hash of a vote $\lambda$, and is digitally signed by $V$. This endorsement asserts that $V$ did not know of equivocations by the sender of $\lambda$ at the time.
+
+A vote $\lambda$ is said to be \emph{endorsed} if it has endorsements by a collection of validators of total weight more than $1/2$ of the total validator weight.
+\end{definition}
+
+The protocol will specify that validators create endorsements only of votes sent by other validators that they do not know to be equivocators. We will discuss later the circumstances under which we will actually need them to create endorsements, but for now, we will assume that they create endorsements for every message they receive by a sender that they do not know to be an equivocator. Once an endorsement is created, it can then be shared among the other validators.
+
+We note that this means that an honest validator will never endorse a pair of equivocating votes as by the time they receive the second of these votes, they will know the sender to be an equivocator and thus will not endorse. In fact, exhibiting endorsements of equivocating messages by a single validator is verifiable proof of bad behavior by that validator for which they can be slashed. Furthermore, assuming that the honest validators make up at least $2/3$rds of the total validator weight, this means that no two endorsed votes can be equivocations. This is because for any two endorsed votes $\lambda_1$ and $\lambda_2$, there must be some honest validator that endorsed both of them.
```

Comment:
> no two endorsed votes can be equivocations

Wouldn't that require `$2/3$` in the definition above, instead of `$1/2$`? It's hard to tell what the right choice here is. I think if you require endorsements by a fraction e, then you need 2e - 1 malicious nodes (by weight) to endorse two equivocating messages; and in general, you need (ke - 1) / (k - 1) malicious nodes to endorse k different chains.

Reply:
You're right, for some reason I was under the impression that 1/2 suffices. It does not..